packer/scripts/macos/macos-agentsetup.sh

```diff
@@ -14,15 +14,27 @@ sudo chown -R ec2-user:staff /var/jenkins
 /usr/local/bin/brew install wget 
 /usr/local/bin/brew install maven
 
-## Install MacPorts, setup java11 and python
+## Install MacPorts, jEnv, setup java8,11,14,17,19,20,21 and python, then set default java to 11
 /usr/local/bin/wget https://github.com/macports/macports-base/releases/download/v2.7.2/MacPorts-2.7.2.tar.gz
 tar -xvf MacPorts-2.7.2.tar.gz
 cd MacPorts-2.7.2
 ./configure && make && sudo make install
 cd .. && rm -rf MacPorts-2.7.2.tar.gz
 export PATH=/opt/local/bin:$PATH
 sudo port -v selfupdate
+yes | sudo port install openjdk8-temurin
 yes | sudo port install openjdk11-temurin
+yes | sudo port install openjdk14-temurin
+yes | sudo port install openjdk17-temurin
+yes | sudo port install openjdk19
+yes | sudo port install openjdk20
+yes | sudo port install openjdk21-temurin
+yes | sudo port install jenv
+echo 'eval "$(jenv init -)"' >> ~/.zshrc && source ~/.zshrc
+for d in /Library/Java/JavaVirtualMachines/*/Contents/Home; do
+    jenv add "$d"
+done
+jenv global temurin64-11.0.21
```

Comment:
Suggestion:
1. Change `d` to some other better variable name to be more visiable
2. See if you can make every entry an alias for easier setup, recommending `openjdk-<majorVersion>` as that is the output of the javaVersion from `detectAgent` jenkins lib such as:

https://github.com/opensearch-project/opensearch-build-libraries/blob/main/vars/detectDockerAgent.groovy#L16C27-L16C34
https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/distribution-build.jenkinsfile#L102

Thanks.